### PR TITLE
Information in Data app filtered by district_id

### PR DESCRIPTION
**Pull request by imsickofmaps**
A District Education Officer needs to only see the data related to Headteachers that they are responsible for. Districts are made up of Zones which have Schools (with an EMIS).
- Django users should have district option
- All the listing views in data app filtered by the district_id value from their user profile

If possible:
- Users with "All" or district_id = 0 should not have filter applied

### CODE REVIEW

File: users/tests.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+"""
```
Comment (imsickofmaps) [inline users/tests.py]: Can delete this please

File: users/views.py [hunk 2]
```diff
@@ -0,0 +1 @@
+# Create your views here.
```
Comment (imsickofmaps) [inline users/views.py]: Delete empty file please

**Comment by imsickofmaps** [comment]
@smn can I have a second pair of eyes on this PR?

File: data/admin.py [hunk 3]
```diff
@@ -10,12 +11,37 @@ class SchoolDataAdmin(admin.ModelAdmin):
...
+        """
+        Limits queries for pages that belong to district admin
+        """
+        qs = super(SchoolDataAdmin, self).queryset(request)
+        if request.user.is_superuser:
+            return qs
+        elif UserDistrict.objects.filter(user_id=request.user.id).exists():
+            return qs.filter(emis__zone__district=request.user.userdistrict.district_id)
+        else:
+            return qs
```
Comment (smn) [inline data/admin.py]: I think we can just return `qs.filter(emis__zone__district=request.user.userdistrict.district_id)` here and avoid the `exists()` all together.
Also please keep an eye out for big joins (like `emis__zone__district`) as these can cause performance problems. I'm not asking you to change this but please use them sparingly and only if really necessary.

File: data/admin.py [hunk 4]
```diff
@@ -10,12 +11,37 @@ class SchoolDataAdmin(admin.ModelAdmin):
...
+        """
+        Limits queries for pages that belong to district admin
+        """
+        qs = super(HeadTeacherAdmin, self).queryset(request)
+        if request.user.is_superuser:
+            return qs
+        elif UserDistrict.objects.filter(user_id=request.user.id).exists():
+            return qs.filter(emis__zone__district=request.user.userdistrict.district_id)
+        else:
+            return qs
```
Comment (smn) [inline data/admin.py]: Same as above, perhaps write a shared base class and make this a bit less wet?

File: data/admin.py [hunk 5]
```diff
@@ -23,6 +49,18 @@ class TeacherPerformanceDataAdmin(admin.ModelAdmin):
...
+        """
+        Limits queries for pages that belong to district admin
+        """
+        qs = super(TeacherPerformanceDataAdmin, self).queryset(request)
+        if request.user.is_superuser:
+            return qs
+        elif UserDistrict.objects.filter(user_id=request.user.id).exists():
+            return qs.filter(emis__zone__district=request.user.userdistrict.district_id)
+        else:
+            return qs
```
Comment (smn) [inline data/admin.py]: ditto.

File: data/admin.py [hunk 6]
```diff
@@ -30,11 +68,36 @@ class LearnerPerformanceDataAdmin(admin.ModelAdmin):
...
+        Limits queries for pages that belong to district admin
+        """
+        qs = super(LearnerPerformanceDataAdmin, self).queryset(request)
+        if request.user.is_superuser:
+            return qs
+        elif UserDistrict.objects.filter(user_id=request.user.id).exists():
+            return qs.filter(emis__zone__district=request.user.userdistrict.district_id)
+        else:
+            return qs
+
```
Comment (smn) [inline data/admin.py:62]: I'm tempted to give you a mop and rip Cmd-V out of your keyboard.

Comment (smn) [inline data/admin.py:62]: Actually, here you go.
mop

File: data/admin.py (hunk 3 above)
Comment (gitumarkk) [inline data/admin.py]: @smn using `request.user.userdistrict.district_id` gives a `DoesNotExist` error if the user doesn't have an associated `district_id`, this error is propagated at the `request.user.userdistrict` level

Comment (smn) [inline data/admin.py]: Ah, I see! Thanks! In that case let's just leave it and apologies for the run-around.

File: users/fixtures/test_hierarchy.json [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+    {
+        "pk": 6,
+        "model": "hierarchy.school",
+        "fields": {
+            "emis": 3633,
+            "name": "Kimale Primary School",
+            "zone": 6
+        }
+    }
+]
```
Comment (imsickofmaps) [inline users/fixtures/test_hierarchy.json]: Missing newline

File: rts/utils.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+                         "TeacherPerformanceDataAdmin", "LearnerPerformanceDataAdmin"]
+            created_by_emis = ["InboundSMSAdmin"]
+            if self.parent.__class__.__name__ in emis_zone:
+                return self.qs.filter(emis__zone__district=self.request.user.userdistrict.district_id)
+            elif self.parent.__class__.__name__ in created_by_emis:
+                return self.qs.filter(created_by__emis__zone__district=self.request.user.userdistrict.district_id)
+            elif self.parent.__class__.__name__ == "SchoolAdmin":
+                return self.qs.filter(zone__district=self.request.user.userdistrict.district_id)
+        else:
+            return self.qs
```
Comment (imsickofmaps) [inline rts/utils.py]: Missing newline

**Comment by imsickofmaps** [comment]
Much DRYer :grinning: :+1:

**Pull request merged**